### PR TITLE
Remove Features put overload

### PR DESCRIPTION
**Pull request by theartful**

### CODE REVIEW

**Comment by karlnapf** [comment]
@gf712 could we have a helper method that applied the read i my prkoert to all parameters for cases like these?

**Comment by gf712** [comment]
sure! we can add something that is called in the base class constructor that adds the readonly mask to everything that is added via SG_ADD

**Comment by karlnapf** [comment]
@theartful I think this would be nicer as then we will never forget to mark parameters as readonly....Something like "mark_class_as_readonly" or so ....

**Comment by gf712** [comment]
@theartful can you also add the m_default_mask to the copy constructor? I forgot it!

**Comment by theartful** [comment]
@gf712 No, you're okay, the copy constructor also calls `init`.

File: tests/unit/ctags [hunk 1]
```diff
@@ -1 +0,0 @@
-/usr/bin/ctags
```
Comment (karlnapf) [inline tests/unit/ctags]: what is this?

File: tests/unit/statistical_testing/KernelSelection_unittest.cc [hunk 2]
```diff
@@ -245,8 +245,8 @@ TEST(KernelSelectionMaxTestPower, quadratic_time_single_kernel)
 
 	auto gen_p=new CMeanShiftDataGenerator(0, dim, 0);
 	auto gen_q=new CMeanShiftDataGenerator(difference, dim, 0);
-	gen_p->CSGObject::put("seed", seed);
-	gen_q->CSGObject::put("seed", seed);
+	gen_p->put("seed", seed);
```
Comment (karlnapf) [inline tests/unit/statistical_testing/KernelSelection_unittest.cc]: not sure I get why this is part of the PR?

File: tests/unit/ctags (hunk 1 above)
Comment (theartful) [reply tests/unit/ctags]: a mistake I made in the RandomMixin PR, thought I drop it now

File: tests/unit/statistical_testing/KernelSelection_unittest.cc (hunk 2 above)
Comment (theartful) [reply tests/unit/statistical_testing/KernelSelection_unittest.cc]: an artifact from the random mixin pr because of the overloaded function

**Comment by theartful** [comment]
let's merge

**Pull request merged**